src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StatelessMarshallingStrategy.cs

```diff
@@ -296,17 +296,16 @@ public IEnumerable<StatementSyntax> GenerateCleanupStatements(TypePositionInfo i
     }
 
     /// <summary>
-    /// Marshaller that enables support for marshalling blittable elements of a collection via a native type that implements the LinearCollection marshalling spec.
+    /// Marshaller type that enables allocating space for marshalling a linear collection using a marshaller that implements the LinearCollection marshalling spec.
     /// </summary>
-    internal sealed class StatelessLinearCollectionBlittableElementsMarshalling : BlittableElementsMarshalling, ICustomTypeMarshallingStrategy
+    internal sealed class StatelessLinearCollectionSpaceAllocator : ICustomTypeMarshallingStrategy
```

Comment:
Is this type ever meant to be used as an ICustomTypeMarshallingStrategy on its own? It looks like it's pretty coupled to the StatelessLinearCollectionMarshalling. Probably not in this PR, but for types like this where it has all the methods of ICustomTypeMarshallingStrategy, but is always supposed to be an "inner marshaller" and isn't meant to be its own the top level ICustomTypeMarshallingStrategy, maybe we should make a different type to make that more clear.

e.g. We could move all the methods from ICustomTypeMarshallingStrategy to a base interface "IMarshallingStatementsGenerator" and have ICustomTypeMarshallingStrategy and ISpaceAllocator both "Marker interfaces" that inherit from it. It might add a little metadata, but I was a bit confused about these scenarios when starting in the codebase.

Reply:
I like that idea. Once this is in, I'll take a look at doing something like that to make it clearer.